src/admin.js

```diff
@@ -1,14 +1,26 @@
 /* global window, fetch */
 /* eslint no-labels: ['error', { 'allowLoop': true }] */
 import 'fetch-everywhere';
-import StitchClient from './client';
+import { _StitchClient } from './client';
 import ADMIN_CLIENT_TYPE from './common';
 import { ADMIN_CLIENT_CODEC } from './auth/common';
 
 const v2 = 2;
 const v3 = 3;
 
-export default class Admin extends StitchClient {
+export class Admin {
+  constructor() {
+    throw new StitchError('Constructor undefined. Use `Admin.init` instead.');
+  }
+
+  static init(baseUrl) {
+    return new Promise((resolve, reject) => {
+      resolve(new _Admin(baseUrl));
```

Comment:
Based on my other comment, I think this can call some "private"/"unexported" method to construct the client. I imagine we'd call `StitchClient.init()`

Reply:
Is this still applicable after our offline conversation?

Reply:
Yes, this will need to call some "constructor" like method that can do async storage prep and then give you back a StitchClient (non-concrete since no interfaces here)